I3uckwheat/cards-of-carousal
# Feature/133 shuffle join code

#### 0. Make sure your issue is linked:

Closes #133 

#### 1. Purpose:

Allow the host to shuffle the lobbies join code.


#### 2. Implementation:


- A new settings sub component was added which sends a dispatch to the host context to shuffle the join code.
- In handling this message the host context sends a message to Lobby in the backend to shuffle the join code.
- Upon receiving this message Lobby executes a callback provided by the LobbyList which handles the actual changing of the join code (which is also the internal lobby id).
- Once LobbyList has made a new code/id and updated the list of lobbies with this new id it calls a function back in Lobby with the new code.
- This function updates the id internally in the lobby and sends a message to the host with the new code.
- When the host receives this message it updates its internal state with the new id/code.


#### 3. Possible Issues:




#### 4. How to Test:


1. Fetch my branch and switch to it
2. Start the application backend then frontend
3. Open one browser window/tab as the host and one as a player
4. Copy the join code and verify that joining the lobby works
5. Click the hamburger
6. Click "SHUFFLE JOIN CODE"
7. Verify that the join code has changed
8. Refresh your player's page so that they leave the lobby
9. Try to join with the original code and verify that it doesn't work
10. Try to join with the new code and verify that it does work

#### 5. Cleanup
